### ISSUE TITLE
Calendar object not found (NextCloud 20.0.5)

### ISSUE THREAD

**Issue by kiplingw** [nextcloud/server#25140]
### Steps to reproduce

I am using NextCloud 2.0.5. On the _Security & setup warnings_ admin panel, I see the following warning: _Your web server is not yet properly set up to allow file synchronisation, because the WebDAV interface seems to be broken._

This is likely related to my calendar, contacts, and all file shares suddenly having stopped working. I believe I noticed the above warning just prior to upgrading from 2.0.? to 2.0.5, but I could be mistaken. I cannot access calendar, contacts, or file shares within the browser for any user, nor remotely via any other WebDAV client (Evolution, NextCloud Desktop Client, nor Android Davx5).

### Expected behaviour
Expected all WebDAV dependent services to continue functioning.

### Actual behaviour
I've spent more than an hour trying to debug unsuccessfully. However, I have managed to isolate a stack trace which does suggest this is WebDAV related:

```json
{
  "reqId": "YAD9DOos62YGfK7Kr9IOvwAAAFQ",
  "level": 0,
  "time": "2021-01-15T02:25:17+00:00",
  "remoteAddr": "<redacted>",
  "user": "<redacted>",
  "app": "webdav",
  "method": "GET",
  "url": "/remote.php/dav/calendars/<redacted>/personal/68290051092cb13e5f191455a74d979f5365416e",
  "message": {
    "Exception": "Sabre\\DAV\\Exception\\NotFound",
    "Message": "Calendar object not found",
    "Code": 0,
    "Trace": [
      {
        "file": "/home/ct_nextcloud/<redacted>/3rdparty/sabre/dav/lib/DAV/Tree.php",
        "line": 80,
        "function": "getChild",
        "class": "OCA\\DAV\\CalDAV\\Calendar",
        "type": "->",
        "args": [
          "68290051092cb13e5f191455a74d979f5365416e"
        ]
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line": 81,
        "function": "getNodeForPath",
        "class": "Sabre\\DAV\\Tree",
        "type": "->",
        "args": [
          "calendars/<redacted>/personal/68290051092cb13e5f191455a74d979f5365416e"
        ]
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "httpGet",
        "class": "Sabre\\DAV\\CorePlugin",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\HTTP\\Request"
          },
          {
            "__class__": "Sabre\\HTTP\\Response"
          }
        ]
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 474,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          "method:GET",
          [
            {
              "__class__": "Sabre\\HTTP\\Request"
            },
            {
              "__class__": "Sabre\\HTTP\\Response"
            }
          ]
        ]
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 251,
        "function": "invokeMethod",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\HTTP\\Request"
          },
          {
            "__class__": "Sabre\\HTTP\\Response"
          }
        ]
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 319,
        "function": "start",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/apps/dav/lib/Server.php",
        "line": 332,
        "function": "exec",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/apps/dav/appinfo/v2/remote.php",
        "line": 35,
        "function": "exec",
        "class": "OCA\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/home/ct_nextcloud/<redacted>/remote.php",
        "line": 167,
        "args": [
          "/home/ct_nextcloud/<redacted>/apps/dav/appinfo/v2/remote.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/home/ct_nextcloud/<redacted>/apps/dav/lib/CalDAV/Calendar.php",
    "Line": 287,
    "CustomMessage": "--"
  },
  "userAgent": "Evolution/3.38.1",
  "version": "20.0.5.2",
  "id": "6000fd0e474a7"
}
```

### Server configuration

**Operating system:** GNU/Linux 4.14.117-grsec-grsec+ x86_64 on AMD Opteron(tm) Processor 4122 (8 cores).

**Web server:** Apache/2.4.29 (Ubuntu)

**Database:** MySQL 5.7.28

**PHP version:** 7.4.11

**Nextcloud version:** 20.0.5.2

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from previous stable release of only a few weeks ago.

**Where did you install Nextcloud from:** Web updater panel.

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.4
  - admin_audit: 1.10.0
  - bruteforcesettings: 2.0.1
  - calendar: 2.1.3
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.3
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.2
  - encryption: 2.8.1
  - federatedfilesharing: 1.10.2
  - federation: 1.10.1
  - files: 1.15.0
  - files_downloadactivity: 1.9.0
  - files_external: 1.11.1
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.2
  - files_trackdownloads: 1.9.0
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - nextcloud_announcements: 1.9.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.3
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - quota_warning: 1.9.1
  - ransomware_protection: 1.8.0
  - recommendations: 0.8.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - user_status: 1.0.1
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - files_antivirus
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.5.2",
        "overwrite.cli.url": "https:\/\/{{ url }}",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "sendmail",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "loglevel": 1,
        "log_type": "file",
        "logfile": "\/home\/***REMOVED SENSITIVE VALUE***\/nextcloud.log",
        "overwriteprotocol": "https",
        "theme": "",
        "encryption.legacy_format_support": false,
        "encryption.key_storage_migrated": false,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** No

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** I don't know.

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
(Apparently disabled.)
```
</details>

### Client configuration
**Browser:** Firefox 84.0.2

**Operating system:** Ubuntu Groovy 20.10

**Comment by kiplingw** [comment]
So I managed to fix this issue. It appears to have nothing to do with NextCloud, but my host's `mod_security` rules.

**Comment by mdkcore0** [comment]
> So I managed to fix this issue. It appears to have nothing to do with NextCloud, but my host's `mod_security` rules.

mind sharing your fix? I've updated to 20.0.4 last week, anything was working until 10 hours ago, when the last file was synced; now just getting that webdav broken error and can't sync anything (desktop or android)

updated to 20.0.5 now, but the problem remains

**Comment by kiplingw** [comment]
Sure @mdkcore0. In case it helps, this is what my server admin wrote me:

> As you figured out, looks as if ModSecurity is flagging it which is causing the issues. I snipped the
errors but this one specifically looks to be saying the issue is xss if you look at the 'Inbound Score':

> [Thu Jan 14 20:46:18.689684 2021] [:error] [pid 15175:tid 4208094787328] [client ***redacted***] [client ***redacted***] ModSecurity: Warning. Operator GE matched 7 at TX:inbound_anomaly_score. [file
"/dh/apache2/template/etc/mod_sec3_CRS/RESPONSE-980-CORRELATION.conf"] [line "87"] [id "980130"] [msg "Inbound Anomaly Score Exceeded (Total Inbound Score: 20 - SQLI=0,XSS=15,RFI=0,LFI=0,RCE=0,PHPI=0,HTTP=0,SESS=0): individual paranoia level scores: 20, 0, 0, 0"] [ver "OWASP_CRS/3.3.0"] [tag "event-correlation"] [hostname "***redacted***"] [uri"/remote.php/dav/calendars/***redacted***/"] [unique_id"YAEeGmH4UiEHwjCbR8O@wAAAABc"]

After that they added an exception and it appears to work now. 

Hopefully that is helpful.
